cache/cache_reservation_manager.h

```diff
@@ -36,6 +36,8 @@ class CacheReservationManager {
   };
   virtual ~CacheReservationManager() {}
   virtual Status UpdateCacheReservation(std::size_t new_memory_used) = 0;
+  virtual Status UpdateCacheReservation(std::size_t memory_used_delta,
```

Comment:
Question:
One annoying thing about the previous `UpdateCacheReservation()` is that it's not multi-thread friendly for `UpdateCacheReservation(GetTotalMemoryUsed() + delta)` as the "update" and the "get memory" is not atomic operation. 

Do you see a way of solving this other than open up another API?

Reply:
Let me look into this later.

Reply:
I like the "Alternatively you could introduce one more CacheReservationManager: a ConcurrentCacheReservationManager" better but this requires changes worth another small PR. Let me TODO and circle back to this during tech debt week.

Reply:
Done

Reply:
Sounds good. Yeah it's probably better to land this this week in time for the release than to block on that cleanup change.